### ISSUE TITLE
Think about supporting Bean Validation 1.1 annotations on fields

### ISSUE THREAD

**Issue by azell** [Randgalt/halva#37]
https://docs.jboss.org/hibernate/stable/beanvalidation/api/
http://beanvalidation.org/

It would be useful to support  JSR-349 annotations on case classes.  Unfortunately, the spec only supports property annotations on JavaBean `getter` methods.  Halva could apply any supplied annotations on the field.

Ex:

```
@CaseClass
class Foo {
  @NotNull
  @Size(min = 8, max = 12)
  String type();
}
```
could generate:
```
class FooCase {
  @NotNull
  @Size(min = 8, max = 12)
  private final String type;
...
```

One implementation is to take the property annotations, filter out `Jackson` annotations and copy the remaining onto the field definition.  See http://fasterxml.github.io/jackson-annotations/javadoc/2.7/com/fasterxml/jackson/annotation/JacksonAnnotation.html:

```
Meta-annotation (annotations used on other annotations) used for marking all annotations that are part of Jackson package. Can be used for recognizing all Jackson annotations generically, and in future also for passing other generic annotation configuration.
```

**Comment by azell** [comment]
From what I remember from the JSR, annotations must be on the field or a `getter` method.  A case class would need to look like:

```
@CaseClass
public interface Foo {
  @NotNull
  String getValue();
}
```

I am not sure if Halva supports the `getXXX` notation or wants to.

**Comment by azell** [comment]
Here is a quick POC implementation with the logic outlined in the initial comment:  https://github.com/azell/halva/commit/faae32dcefeac3213b0c941798e04e25e48d9adc

I am not totally in love with this approach, and would welcome alternate viewpoints.

**Comment by Randgalt** [comment]
Fixed by #40
